lib/graphql/formatter/adapter.ex

```diff
@@ -0,0 +1,34 @@
+defmodule BridgeEx.Graphql.Formatter.Adapter do
+  @moduledoc """
+  Behaviour used to implement a formatter adapter which could be used to
+  format types for both input variables and responses.
+
+  ```elixir
+  defmodule MyCustomAdapter do
+
+    @behaviour BridgeEx.Graphql.Formatter.Adapter
+
+    def format(variable) when is_map(variable) do
+      variable
+      |> Enum.map(fn
+        {key, value} -> {key, format(value)}
+      end)
+      |> Map.new()
+    end
+
+    def format(variable = %Date{}), do: Date.to_string(variable)
+  end
+
+  defmodule MyBridge do
+    use BridgeEx.Graphql,
+      format_variables_with: MyCustomAdapter
+  end
+  ```
+  """
+
+  @type t() :: %__MODULE__{}
+
+  defstruct name: __MODULE__
+
+  @callback format(any()) :: any()
```

Comment:
Per facilitare la scrittura di eventuali adapter, potrebbe avere senso definire la `format` qui dentro come e' definita sotto, visto che alla fine tendenzialmente si itera su una mappa o simili, e invece definire due callback

- format_key
- format_value

Reply:
Non sono molto convinto, avere due callback, una per la key ed una per il value sarebbe un po' limitante.
Esempio estremo: qualcuno deve agire su un value in funzione della key. Non potrebbe farlo perché nella format_value non avrebbe alcuna informazione riguardo alla key.